internal/db_info.go

```diff
@@ -0,0 +1,112 @@
+// Copyright © 2025 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/jackc/pgx/v5"
+)
+
+// DbInfo provides information about tables in a database.
+type DbInfo struct {
+	conn  *pgx.Conn
+	cache map[string]*tableCache
+}
+
+// tableCache stores information about a table.
+// The information is cached and refreshed every 'cacheExpiration'.
+type tableCache struct {
+	columns     map[string]int
+	nextRefresh time.Time
+}
+
+// cacheExpiration defines how long table information remains valid
+const cacheExpiration = 5 * time.Minute
```

Comment:
Could we make this configurable instead? To let the user control this.

Reply:
Hmm, that's an option too. However, I don't know if a user would find this useful or what would drive their decision. Schema changes are very infrequent, and you can't really predict them, which means that configuring them is also likely to be just guessing.

Reply:
I don't think this needs to be exposed to the user.